tokio/src/net/unix/listener.rs

```diff
@@ -70,7 +74,20 @@ impl UnixListener {
     where
         P: AsRef<Path>,
     {
-        let listener = mio::net::UnixListener::bind(path)?;
+        // For now, we handle abstract socket paths on linux here.
+        #[cfg(target_os = "linux")]
+        let addr = {
+            let os_str_bytes = path.as_ref().as_os_str().as_bytes();
+            if os_str_bytes.starts_with(b"\0") {
+                StdSocketAddr::from_abstract_name(os_str_bytes)?
+            } else {
+                StdSocketAddr::from_pathname(path)?
+            }
+        };
```

Comment:
Is Linux the only OS that needs this fix? Does macOS not have abstract path names? It would be worth to look at how mio did this prior to the v1 release.

Reply:
> Is Linux the only OS that needs this fix?

According to my quick research, yes.
Actually `SocketAddrExt::from_abstract_name` is available on linux only.
* https://doc.rust-lang.org/stable/std/os/linux/net/trait.SocketAddrExt.html
* https://github.com/rust-lang/rust/issues/85410

It seems that mio previously used its own parser to handle all unix systems.
* https://github.com/tokio-rs/mio/blob/cf5219446de5190752372f5401c51987248406f5/src/sys/unix/uds/mod.rs#L25-L72

Reply:
Ah, ok. Does `target_os = "linux"` also include Android? I see that it's also available there.

Reply:
Yes, we should probably add `target_os = "android"` as well. I just added now.